src/gui/MenuBar.cpp

```diff
@@ -466,6 +494,89 @@ void MenuBar::setKeyShortcuts()
 #endif
 }
 
+void MenuBar::changeProfile()
+{
+	QAction *act = (QAction *)sender();
+	const QString selectedProfile = act->property("path").toString();
+	QSettings profileSettings(QMPlay2Core.getSettingsDir() + "Profile.ini", QSettings::IniFormat);
+	if (selectedProfile != profileSettings.value("Profile", "/").toString())
+	{
+		profileSettings.setValue("Profile", selectedProfile);
+		QMPlay2GUI.restartApp = true;
+		QMPlay2GUI.mainW->close();
+	}
+}
+void MenuBar::addProfile()
+{
+	const QString selectedProfile = Functions::cleanFileName(QInputDialog::getText(
+				this, Options::tr("Create new profile"),
+				Options::tr("Profile name")));
+	if (selectedProfile.isEmpty())
+		return;
+	QSettings profileSettings(QMPlay2Core.getSettingsDir() + "Profile.ini", QSettings::IniFormat);
+	if (selectedProfile != profileSettings.value("Profile", "/").toString())
+	{
+		QDir(QMPlay2Core.getSettingsDir()).mkpath("Profiles/" + selectedProfile);
+		QFile tempFile(QMPlay2Core.getSettingsDir() + "Profiles/" + selectedProfile + "/QMPlay2.ini");
+		tempFile.open(QFile::WriteOnly);
+		tempFile.close();
+
+		profileSettings.setValue("Profile", selectedProfile);
+		QMPlay2GUI.restartApp = true;
+		QMPlay2GUI.mainW->close();
+	}
+}
+void MenuBar::copyProfile()
```

Comment:
`copyProfile()` and `addProfile()` can share the same code e.g. in `static bool ...()` function.